udmtests/udm_nf_discovery_test.go

```diff
@@ -0,0 +1,633 @@
+// SPDX-FileCopyrightText: 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * UDM Unit Testcases
+ *
+ */
+package udmtests
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/antihax/optional"
+	"github.com/omec-project/openapi/Nnrf_NFDiscovery"
+	"github.com/omec-project/openapi/models"
+	"github.com/omec-project/udm/consumer"
+	udmContext "github.com/omec-project/udm/context"
+	"github.com/omec-project/udm/factory"
+	"github.com/omec-project/udm/producer"
+	"github.com/omec-project/udm/service"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	UDMTest        = &service.UDM{}
+	nfInstanceID   = "34343-4343-43-434-343"
+	subscriptionID = "46326-232353-2323"
+)
+
+func setupTest() {
+	if err := factory.InitConfigFactory("../factory/udmcfg.yaml"); err != nil {
+		fmt.Printf("Could not InitConfigFactory: %+v\n", err)
+	}
+}
+
+func TestCheckNRFCachingIsEnabled(t *testing.T) {
+	got := factory.UdmConfig.Configuration.EnableNrfCaching
+	assert.Equal(t, got, true, "NRF Caching is not enabled.")
+}
+
+func TestRegisterNF(t *testing.T) {
+	origRegisterNFInstance := consumer.SendRegisterNFInstance
+	origSearchNFInstances := consumer.SendSearchNFInstances
+	origUpdateNFInstance := consumer.SendUpdateNFInstance
+	defer func() {
+		consumer.SendRegisterNFInstance = origRegisterNFInstance
+		consumer.SendSearchNFInstances = origSearchNFInstances
+		consumer.SendUpdateNFInstance = origUpdateNFInstance
+	}()
+	fmt.Printf("test case TestRegisterNF \n")
+	var prof models.NfProfile
+	consumer.SendRegisterNFInstance = func(nrfUri string, nfInstanceId string, profile models.NfProfile) (models.NfProfile, string, string, error) {
```

Comment:
The [gock](https://github.com/h2non/gock) can help to test the consumer handlers, for example:
https://github.com/free5gc/udm/blob/main/internal/sbi/consumer/nf_management_test.go#L16

Reply:
Hello @ianchen0119,

I checked this testing library and it is not getting contribution for last two year. I think it would be better to continue without using this library.